geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/legacy/attribute/AttributeQueryableIndex.scala

```diff
@@ -235,12 +236,23 @@ trait AttributeQueryableIndex extends AccumuloFeatureIndex with LazyLogging {
       attributeScan.columnFamilies, recordThreads, hasDupes, joinFunction, joinQuery)
   }
 
-  override def getFilterStrategy(sft: SimpleFeatureType, filter: Filter): Seq[AccumuloFilterStrategyType] = {
+  override def getFilterStrategy(sft: SimpleFeatureType,
+                                 filter: Filter,
+                                 transform: Option[SimpleFeatureType]): Seq[AccumuloFilterStrategyType] = {
+    import AttributeIndex.AllowJoinPlans
     val attributes = FilterHelper.propertyNames(filter, sft)
-    val indexedAttributes = attributes.filter(a => Option(sft.getDescriptor(a)).exists(_.isIndexed))
-    indexedAttributes.flatMap { attribute =>
+    val indexedAttributes = attributes.flatMap { a =>
+      Option(sft.getDescriptor(a)).map(_.getIndexCoverage()).toSeq.collect {
+        case coverage if coverage != IndexCoverage.NONE => (a, coverage)
+      }
+    }
+    indexedAttributes.flatMap { case (attribute, coverage) =>
       val (primary, secondary) = FilterExtractingVisitor(filter, attribute, sft, attributeCheck)
-      if (primary.isDefined) {
+      lazy val joinCheck = coverage match {
+        case IndexCoverage.FULL => true
+        case IndexCoverage.JOIN => AllowJoinPlans.get || !requiresJoin(sft, attribute, secondary, transform)
+      }
+      if (primary.isDefined && joinCheck) {
```

Comment:
Just so I'm clear, ``joinCheck`` should return true IFF the FilterStrategy doesn't involve a join OR if the caller explicitly says join plans are OK? I had to read this a couple of times to get that. (That may be a me problem more than a problem with your PR.)

It would be nice (i.e. more readable?) if the logic for this wasn't split between here and the ``requiresJoin`` method, but I know you also need that functionality in ``getCost``, so this might be the best solution.